bgpd/bgp_mplsvpn.c

```diff
@@ -4074,6 +4074,32 @@ void bgp_vpn_leak_export(struct bgp *from_bgp)
 	}
 }
 
+void bgp_vpn_release_label(struct bgp *bgp, afi_t afi, bool reset)
+{
+	if (!CHECK_FLAG(bgp->vpn_policy[afi].flags, BGP_VPN_POLICY_TOVPN_LABEL_AUTO))
+		return;
+	/*
+	 * label has previously been automatically
+	 * assigned by labelpool: release it
+	 *
+	 * NB if tovpn_label == MPLS_LABEL_NONE it
+	 * means the automatic assignment is in flight
+	 * and therefore the labelpool callback must
+	 * detect that the auto label is not needed.
+	 */
+	if (bgp->vpn_policy[afi].tovpn_label == MPLS_LABEL_NONE)
+		return;
+	if (CHECK_FLAG(bgp->vpn_policy[afi].flags, BGP_VPN_POLICY_TOVPN_LABEL_PER_NEXTHOP))
+		return;
+
+	bgp_lp_release(LP_TYPE_VRF, &bgp->vpn_policy[afi], bgp->vpn_policy[afi].tovpn_label);
+
+	if (reset) {
```

Comment:
don't follow this: why release a label, but leave it in `tovpn_label` ?

Reply:
This is to avoid doing unnecessary changes because this function is reused in vty to change the allocation-mode.
This said, I think it can be acceptable to move:
```
                        bgp->vpn_policy[afi].tovpn_label = MPLS_LABEL_NONE;                                                                 
```
just after 
```bgp_lp_release```

Reply:
moved  **bgp->vpn_policy[afi].tovpn_label = MPLS_LABEL_NONE;**